packages/app-webdir-ui/src/helpers/dataConverter.js

```diff
@@ -105,106 +105,97 @@ const fillInBlanks = datum => {
   };
   return { ...full, ...datumAdjusted };
 };
+function isCustomTitle(profile, index) {
+  if (!profile.title_source?.raw) return false;
+  return profile.title_source?.raw[index] === "titles"
+}
 
-// If it's depts only - use that dept in place of primary_deptid
+function findDeptIndex(profile, deptId = "") {
+  if (!deptId) return 0;
+  return profile.deptids?.raw?.findIndex(id => id === deptId);
+}
 
-const getTitleFromProfile = (profile, titleMatch) => {
-  // Note on title logic
-  // - Engine WEB_DIRECTORY_PEOPLE_AND_DEPS supplies results with title logic
-  //   already handled by the service.
-  // - Engine WEB_DIRECTORY_DEPARTMENTS needs a little help with titles,
-  //   handled below using titleMatch.depts.
+function getTitleAndDeptFromIndex(profile, index) {
+  let title = profile.titles?.raw[index];
+  let dept = profile.departments?.raw[index];
+  if (profile.title_source.raw[index] !== "titles" && profile.working_title?.raw[0]) {
+    title = profile.working_title.raw[0];
```

Comment:
This is doing a fallback to `working_title` when the `title_source` doesn't identify a custom title. ... This would only be okay as long as it's only used when there's no dept provided but because we have the index from the department here... 

I only see it being used for the Courtesy Affiliate logic flow... if this is unique to that case, It would be better to change the name of the function or to embed the logic in the code, otherwise this runs the risk of causing confusion at a later date.

BUT, after setting up a test in the Test environment, and setting the `WebDirectoryExampleDepartments` story so it uses `deptIds="1342"` (ASU department) it display's Audrey's record - and this is not her primary affiliation, but it's displaying her HR title (working title) while it should display no title. So more work is needed to ensure we match the logic outlined in https://docs.google.com/document/d/1ypPuf-zmEPpKLpmry3paQTsQ5uaxX-_l1IjCHANwykU/edit .

Reply:
That's convoluted. In essence, when we're displaying a profile and the department is supplied, we only should fallback to the working title if the deptId matches the person's primary_deptid. ... There may be some wiggly-ness there, so maybe we can touch base with @tbutterf and align on approach.